server/routes/filmsRoutes.js

```diff
@@ -0,0 +1,52 @@
+const mongoose = require('mongoose');
+const Film = mongoose.model('films');
+const User = mongoose.model('users');
+
+module.exports = (app) => {
+  app.get(`/api/films/:pageId/`, async (req, res) => {
+    const { pageId } = req.params
+    i = 5
+    y = i * (pageId - 1)
+    const films = await Film.find().skip(y).limit(i);
+
+    return res.status(200).send(films);
+  });
+
+  app.get(`/api/film/title`, async (req, res) => {
+    const films = await Film.find({ "title": { "$regex": req.query.keyword }});
+
+    return res.status(200).send({
+      error: false,
```

Comment:
We are returning error false in every one of our responses, but we never set it as true, even when there are actual failures.

Could we handle those cases, especially consider what we want to do when we can't find the film or could not connect to the database

Reply:
We can do something simple like 
```javascript
try {
    let data = await Film.find(condition, constraints ? constraints : null);
    console.log(`findOne success--> ${data}`);
    if(!data) {
      throw new Error('no document found');
    }
    return data;
} catch (error) {
    error = true
    return error;
}
```

It might also be worth exploring other alternatives through using mongoose's [orFail](https://mongoosejs.com/docs/api.html#query_Query-orFail) functions or just generally error handling in [nodejs](https://stackify.com/node-js-error-handling/)